numba/core/types/functions.py

```diff
@@ -628,6 +628,10 @@ def generic(self, args, kws):
     def key(self):
         return self.sig, self.cconv, self.get_pointer
 
+    def unify(self, typingctx, other):
+        if isinstance(other, Function):
+            return self
+
```

Comment:
```suggestion
```
@DrTodd13 xref: https://github.com/numba/numba/pull/7625#issuecomment-1164307865 suggest removing this.